src/matchers.ts

```diff
@@ -67,7 +65,10 @@ class IsInStorylet implements StateMatcher {
     }
 
     match(state: GameState): boolean {
-        return state.storyletPhase == StoryletPhases.In && state.storyletId === this.expectedStoryletId;
+        if (state.storyletPhase == StoryletPhases.In && 'id' in state.currentStorylet) {
```

Comment:
`state.storyletPhase == StoryletPhases.In` could probably be removed, as result storylets seem to have an ID separate from their root storylets (is this always the case?). Then the predicate can be used to match `In` or `End` storylets at will just by ID.

Reply:
After playing around with some storylets at Marigold I tentatively agree that simply matching by ID should be enough here.

Reply:
Great. I've removed that part from the clause.